ci/workflows/pull-request.yml

```diff
@@ -63,7 +63,8 @@ steps:
           run: rw-build-env
           config: ci/docker-compose.yml
           mount-buildkite-agent: true
-    timeout_in_minutes: 15
+    # FIXME: decrease this when we found what's causing the build to be slow.
```

Comment:
Investigate why sim build becomes slower

and decrease timeouts

<img width="1349" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/37948597/8eb812be-8f63-475a-a0a0-c0e788ef5821">